backend/core/src/filters/catch-all.filter.ts

```diff
@@ -0,0 +1,15 @@
+import { Catch, ArgumentsHost } from "@nestjs/common"
+import { BaseExceptionFilter } from "@nestjs/core"
+
+@Catch()
+export class CatchAllFilter extends BaseExceptionFilter {
+  catch(exception: any, host: ArgumentsHost) {
+    console.error({ message: exception?.response?.message, stack: exception.stack, exception })
+    if (exception.name === "EntityNotFound") {
```

Comment:
I think it will never be triggered because as per NestJS docs:

> The @Catch(HttpException) decorator binds the required metadata to the exception filter, telling Nest that this particular filter is looking for exceptions of type HttpException and nothing else. The @Catch() decorator may take a single parameter, or a comma-separated list. This lets you set up the filter for several types of exceptions at once.

and 

```
@Catch(EntityNotFoundError)
export class EntityNotFoundExceptionFilter implements ExceptionFilter {
  catch(exception: EntityNotFoundError, host: ArgumentsHost) {
    const response = host.switchToHttp().getResponse()
    response.status(404).json({ message: exception.message })
  }
}
```

returns a response and by that short circuits filters chain, so CatchAllFilter will never be called when EntityNotFoundError is thrown. 

Could you rephrase that part: 

>I rolled logic into here from the entity not found exception filter, but in a second pass we should remove the exception filter specific to it and possibly remove this custom logic all together if its not needed.

What do you want to do with `EntityNotFoundExceptionFilter`?

Reply:
Hey @pbn4 

for `I think it will never be triggered because as per NestJS docs:` peep https://docs.nestjs.com/exception-filters#catch-everything

when you don't pass an exception list in it catches everything

for `so CatchAllFilter will never be called when EntityNotFoundError is thrown`
I thought so to, but weirdly there was an e2e test failing where the EntityNotFoundError was only getting caught in the new catch all filter 

also for `Could you rephrase that part:` sorry, I meant that with the catch all filter in place we no longer need that specific filter anymore so we can get rid of the specific filter for it code wise. I thought maybe in the future we could get rid of that entirely and let nest's default exception list cover it for us. 

Lemme know if you have any other questions or anything like that

Reply:
> for so CatchAllFilter will never be called when EntityNotFoundError is thrown
I thought so to, but weirdly there was an e2e test failing where the EntityNotFoundError was only getting caught in the new catch all filter

OK I see what the problem was, the `EntityNotFoundExceptionFilter` is never called because TypeORM does not throw a proper EntityNotFoundError class instance, it throws a JS object. A proper implementation of this filter should be 
```
@Catch(NotFoundException)
export class NotFoundExceptionFilter implements ExceptionFilter {
  catch(exception: NotFoundException, host: ArgumentsHost) {
    const response = host.switchToHttp().getResponse()
    response.status(404).json({ message: exception.message })
  }
}
```

because most of the services use findOne and check whether returned object is undefined and then throw NotFoundException. 

Anyway your approach is better so let's keep that, you can remove `EntityNotFoundExceptionFilter` completely in this PR.